.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+name: Test Plugin
+
+on:
+  push:
+    branches-ignore:
+      - 'whitesource-remediate/**'
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        test:
+          - test jacocoTestReport
+          - integTest
+          - yamlRestTest
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        java:
+          - 17
```

Comment:
I remember in AD we had to test on 11, 17, 20. Should we do that here too? I think default for 2.x is jdk 11 still but could be wrong

Reply:
@dbwiddis WDYT? Since you worked on the CI initially.

Reply:
https://opensearch.org/docs/latest/install-and-configure/install-opensearch/index/#java-compatibility this is what we have documented and i think java 20 is going to be for 3.x

Reply:
Pushed the change.

Reply:
We should probably match whatever OpenSearch tests.